executor/src/main/java/org/k8loud/executor/openstack/OpenstackNeutronServiceImpl.java

```diff
@@ -23,22 +23,47 @@ public void createSecurityGroup(String name, String description,
                 .description(description)
                 .build());
 
-        if (securityGroup == null){
+        if (securityGroup == null) {
             throw new OpenstackException(CREATE_SECURITY_GROUP_FAILED,
                     "Failed to create SecurityGroup with name \"%s\"", name);
         }
     }
 
     @Override
-    public SecurityGroup getSecurityGroup(String securityGroupId, OSClient.OSClientV3 client) throws OpenstackException {
+    public SecurityGroup getSecurityGroup(String securityGroupId,
+                                          OSClient.OSClientV3 client) throws OpenstackException {
         log.debug("Getting security group object with id '{}'", securityGroupId);
         SecurityGroup securityGroup = client.networking().securitygroup().get(securityGroupId);
 
-        if (securityGroup == null){
+        if (securityGroup == null) {
             throw new OpenstackException(SECURITY_GROUP_NOT_EXIST,
                     "Failed to find SecurityGroup with '%s'", securityGroupId);
         }
 
         return securityGroup;
     }
+
+    @Override
+    public SecurityGroupRule createSecurityGroupRule(SecurityGroup securityGroup, String ethertype, String direction,
+                                                     String remoteIpPrefix, String protocol, int portRangeMin,
+                                                     int portRangeMax,
+                                                     OSClient.OSClientV3 client) throws OpenstackException {
+        log.debug("Creating new rule for SecurityGroup with id '{}'", securityGroup.getName());
+        SecurityGroupRule securityGroupRule = client.networking().securityrule().create(Builders.securityGroupRule()
+                .securityGroupId(securityGroup.getId())
+                .direction(direction)
+                .ethertype(ethertype)
+                .portRangeMin(portRangeMin)
+                .portRangeMax(portRangeMax)
+                .protocol(protocol)
+                .remoteIpPrefix(remoteIpPrefix)
+                .build());
+
+        if (securityGroupRule == null) {
+            throw new OpenstackException(ADD_RULE_FAILED,
```

Comment:
You're mixing names; once it's `add` (exception code), the other time it's `create` (method name), stick to one of the two

Reply:
changed